vlasovsolver/cpu_acc_map.cpp

```diff
@@ -132,6 +132,8 @@ bool map_1d(SpatialCell* spatial_cell,
    if(vmesh.size() == 0 )
       return true;
 
+   const int NUM_ASYNC_QUEUES=P::openaccQueueNum;
+   const int openacc_async_queue_id = (int)(spatial_cell->parameters[CellParams::CELLID]) % NUM_ASYNC_QUEUES;
```

Comment:
Hm, there's no guarantee that our cellIDs would be continuous per single task, so this could end up with weird slowdowns...

Reply:
And indeed, it does. The nsight graph shows occasional syncpoints, where subsequent cells are submitted to the same queue.

I wonder though: now that the calls to this function are wrapped in a loop over cells, do we even *need* the wait call inside map_1d anymore? Or can we just happily stuff more work into the same async queue to be evaluated at some later time? The overall wait statement at the end of each acceleration order should take care of synchronization then.

Reply:
sounds reasonable

Reply:
A quick test shows that the wait statement here can be tossed right out, without any negative effects. It seems faster too, but I'd want to see that quantitatively checked out before I commit anything.

Reply:
Yep, I agree that if we have a pragma acc wait at each dimension order loop, we can lose the wait within map_1d. In this case, we're using the async queues in such a manner that we could just use the CellID as the queue ID. This could be tested by setting an arbitrarily large value to P::openaccQueueNum.